### PR TITLE
porting diag_second_moments_srf function

### PR DESCRIPTION
**Pull request by jgfouca**

### CODE REVIEW

**Comment by jgfouca** [comment]
@xyuan , my apologies, I had to force-push master to revert the merge of #501 . That PR got merged before it was ready.

**Comment by jgfouca** [comment]
I will review soon with the identified problems.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 1]
```diff
@@ -46,11 +46,43 @@ void eddy_diffusivities_c(Int nlev, Int shcol, Real *obklen, Real *pblh,
...
+
 }
 
 namespace scream {
 namespace shoc {
 
+namespace {
+
+template <size_t N, size_t M>
+void gen_random_data(const std::array<std::pair<Real, Real>, N>& ranges,
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:60]: Please use the randomize() method off of the base class. If you need a random range other than 0->1, I'm sure you can figure how to do this by adding some optional arguments to randomize.

File: components/scream/src/physics/shoc/shoc_functions_f90.hpp [hunk 2]
```diff
@@ -180,6 +180,25 @@ struct SHOCVarorcovarData : public SHOCDataBase {
   { SHOCDataBase::operator=(rhs); tunefac = rhs.tunefac; return *this; }
 };//SHOCVarorcovarData
 
+struct MomSrfData {
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.hpp:183]: Please follow the pattern of the other Data structs in this file.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 3]
```diff
@@ -256,5 +322,39 @@ void calc_shoc_vertflux_f(Int shcol, Int nlev, Int nlevi, Real *tkh_zi,
...
+
+  pack::host_to_device({wthl, uw, vw, ustar2, wstar}, shcol, mom_d);
+
+  view_1d wthl_d(mom_d[0]),
+          uw_d(mom_d[1]),
+          vw_d(mom_d[2]),
+          ustar2_d(mom_d[3]),
+          wstar_d(mom_d[4]);
+
+  auto policy = util::ExeSpaceUtils<ExeSpace>::get_default_team_policy(1, nshcol_pack);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:345]: We won't be packing data along shcol.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 4]
```diff
@@ -256,5 +322,39 @@ void calc_shoc_vertflux_f(Int shcol, Int nlev, Int nlevi, Real *tkh_zi,
...
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
+
+    SHOC::shoc_diag_second_moments_srf(team, shcol, wthl_d, uw_d, vw_d, ustar2_d, wstar_d);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:348]: Change to pass single packs, not views.

File: components/scream/src/physics/shoc/tests/shoc_diag_second_mom_srf_test.cpp [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  for (Int i = 0; i < num_runs; ++i) {
+    MomSrfData& d = srf_cxx[i];
+    shoc_diag_second_moments_srf_f(d.shcol, d.wthl, d.uw, d.vw, d.ustar2, d.wstar);
+  }
+
+
+  for (Int i = 0; i < num_runs; ++i) {
+    Int shcol = srf_cxx[i].shcol;
+    for (Int k = 0; k < shcol; ++k) {
+      REQUIRE(srf_fortran[i].wthl[k]   == srf_cxx[i].wthl[k]);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/tests/shoc_diag_second_mom_srf_test.cpp:73]: You only need to check outputs here.

**Comment by jgfouca** [comment]
@xyuan I can't change you to be the author, so I cannot review the PR (since I am the author). Please see my comments, a fair bit of work is needed on this PR.

**Comment by xyuan** [comment]
ok, I am closing this PR and create a new one.

**Pull request closed without merging**